pkg/pkg.bzl

```diff
@@ -312,9 +312,12 @@ def pkg_tar(**kwargs):
                       "This attribute was renamed to `srcs`. " +
                       "Consider renaming it in your BUILD file.")
                 kwargs["srcs"] = kwargs.pop("files")
+    archive_name = kwargs.get("archive_name") or name
```

Comment:
I'm torn over the best name for this attribute since the common word for tar it is archive; for zip, file; for .deb/.rpm, package.  But having universal word for the concept is better than being specific for each type. 

How does `package_name` sound?  Maybe pkg_name?

Reply:
Regarding the name of the attribute, I struggled a bit with that, and I was interested to hear what you'd prefer to call it. I considered both `pkg_name` and `package_name`. I had a hard time deciding between them, because this library uses the abbreviation in some places and the full word in other places. Since `pkg` seems to be used for the names of rules and macros whereas `package` seems to be used for attribute names, I'd think the latter may be preferable for consistency. Ultimately I settled on `archive_name`, though, because the word "package" is heavily overloaded in this space. More specifically, `pkg_deb` already has a parameter called `package` that means something slightly different, and I thought it would be confusing for that macro to accept both `package` and `package_name`. What do you think?

Yes, I'm aware of #193. I work with Konstantin (who opened that issue), and we've discussed his proposal. Konstantin suggested the macros could be written so that it would be possible either to specify the package/archive name (and `package_dir`) directly or to have them generated via a provider.